tests/process_manager/views/test_partial_views.py

```diff
@@ -59,15 +59,21 @@ class TestMessagesView(LoginRequiredTest):
     """Test the process_manager.views.messages view function."""
 
     endpoint = reverse("process_manager:messages")
+    topic = "control.test.process_manager"
+
+    @pytest.fixture(autouse=True)
+    def kafka_topic_regex(self, settings):
```

Comment:
I haven't been following the Kafka stuff. Why do we need to use regex? The DUNE team keep talking about it like it's a structured object, not just a random string. Surely there's a way to avoid dodgy string manipulations?

Reply:
Unfortunately the only way to avoid regex is to know the entire list of topics upfront, and define each one of them explicitly as a `topics` list in [`subscribe()`](https://kafka-python.readthedocs.io/en/master/apidoc/KafkaConsumer.html#kafka.KafkaConsumer.subscribe).

Procman topics have the form `control.<session>.process_manager`, and there's a lot of them. Defining as regex also has the advantage that the Kafka listener periodically refreshes the topics it listens to, so new topics can be dynamically picked up if they are added later and match the pattern.